File: app/src/main/java/ru/iandreyshev/parserrss/models/rss/ViewArticle.java
package ru.iandreyshev.parserrss.models.rss;

import android.support.annotation.NonNull;

import ru.iandreyshev.parserrss.models.repository.Article;

public class ViewArticle {
    private final long mId;
    private final String mTitle;
    private final String mOriginUrl;
    private final String mDescription;
    private final Long mDate;

    public ViewArticle(@NonNull final Article article) {
        mId = article.getId();
        mTitle = article.getTitle();
        mDescription = article.getDescription();
        mOriginUrl = article.getOriginUrl();
        mDate = article.getDate();
    }

    public long getId() {
        return mId;
    }

    @NonNull
    public String getTitle() {
        return mTitle;
    }

    @NonNull
    public String getDescription() {
        return mDescription;
    }

    public String getOriginUrl() {
        return mOriginUrl;
    }

    public Long getDate() {
        return mDate;
    }

    @Override
    public final boolean equals(Object other) {
        return (other instanceof ViewArticle) && mOriginUrl.equals(((ViewArticle) other).mOriginUrl);
    }

    @Override
    public final int hashCode() {
        return mOriginUrl.hashCode();
    }
}

File: app/src/main/java/ru/iandreyshev/parserrss/models/rss/ViewArticle.kt
package ru.iandreyshev.parserrss.models.rss

import ru.iandreyshev.parserrss.models.repository.Article

class ViewArticle(article: Article) {
    val id: Long = article.id
    val title: String = article.title
    val originUrl: String = article.originUrl
    val description: String = article.description
    val date: Long? = article.date

    override fun equals(other: Any?): Boolean {
        return other is ViewArticle && originUrl == other.originUrl
    }

    override fun hashCode(): Int {
        return originUrl.hashCode()
    }
}
